packages/web/hooks/use-swap.ts

```diff
@@ -81,7 +85,50 @@ export function useSwap({
     useOtherCurrencies,
   });
 
-  const inAmountInput = useAmountInput(swapAssets.fromAsset);
+  const [gasAmount, setGasAmount] = useState<CoinPretty>();
+  const inAmountInput = useAmountInput({
+    currency: swapAssets.fromAsset,
+    gasAmount: gasAmount,
+  });
+
+  const {
+    data: maxBalanceQuote,
```

Comment:
Not clear why `maxBalanceQuote` means? I am not seeing in the input anything to suggest that the max balance is passed as argument to the hook.

Reply:
`maxBalanceQuote` is a quote using the account's max balance for tokenIn. In line [102](https://github.com/osmosis-labs/osmosis-frontend/pull/3072/files/3c7a635f8d2412d9b2dc0453ac10379b95594a89#diff-188329441517b71d6811aaaf32971da03955042040446222fe5fb397220e5bf9R102), we get the max balance from the `useAmountInput` hook. 

Perhaps naming could be better, any ideas?

Reply:
currently it's called [quote](https://github.com/osmosis-labs/osmosis-frontend/blob/9acea4f0af4b841843c1717fed007b80696472d2/packages/web/hooks/use-swap.ts#L98C11-L98C16). It's a results of a query hook `useQueryRouterBestQuote` where the in amount value is `inAmountInput.balance?.toCoin().amount!` nothing else here tells me we are actually working with a `max value`, so it is confusing, and looks like a mistake, or it makes the expectation for a reader to know something they might not know. Maybe a better name is `quoteForCurrentBalance`?